dotCMS/src/main/java/com/dotcms/rest/api/v1/content/VersionResource.java

```diff
@@ -0,0 +1,249 @@
+
+import static com.dotcms.rest.api.v1.authentication.ResponseUtil.getFormattedMessage;
+
+import com.dotcms.repackage.javax.ws.rs.GET;
+import com.dotcms.repackage.javax.ws.rs.Path;
+import com.dotcms.repackage.javax.ws.rs.PathParam;
+import com.dotcms.repackage.javax.ws.rs.Produces;
+import com.dotcms.repackage.javax.ws.rs.QueryParam;
+import com.dotcms.repackage.javax.ws.rs.core.Context;
+import com.dotcms.repackage.javax.ws.rs.core.MediaType;
+import com.dotcms.repackage.javax.ws.rs.core.Response;
+import com.dotcms.repackage.org.glassfish.jersey.server.JSONP;
+import com.dotcms.rest.InitDataObject;
+import com.dotcms.rest.ResponseEntityView;
+import com.dotcms.rest.WebResource;
+import com.dotcms.rest.annotation.NoCache;
+import com.dotcms.rest.api.v1.authentication.ResponseUtil;
+import com.dotcms.rest.exception.BadRequestException;
+import com.dotcms.uuid.shorty.ShortType;
+import com.dotcms.uuid.shorty.ShortyId;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.exception.DoesNotExistException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.contentlet.transform.ContentletTransformerJson;
+import com.dotmarketing.portlets.languagesmanager.business.LanguageAPI;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.util.Logger;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import com.liferay.portal.model.User;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletRequest;
+
+@Path("/v1/versions")
+public class VersionResource {
+
+    public static final int MIN = 20;
+    public static final int MAX = 100;
+
+    private final WebResource webResource;
+    private final ContentletAPI contentletAPI;
+    private final LanguageAPI languageAPI;
+
+
+    public VersionResource() {
+        this(APILocator.getContentletAPI(), APILocator.getLanguageAPI(), new WebResource());
+    }
+
+    @VisibleForTesting
+    public VersionResource(final ContentletAPI contentletAPI, final LanguageAPI languageAPI,
+            final WebResource webResource) {
+        this.contentletAPI = contentletAPI;
+        this.languageAPI = languageAPI;
+        this.webResource = webResource;
+    }
+
+
+    @GET
+    @JSONP
+    @NoCache
+    @Path("/all/{identifier}")
+    @Produces(MediaType.APPLICATION_JSON + ";charset=UTF-8")
+    public Response findAllVersions(@Context final HttpServletRequest request,
+            @PathParam("identifier") final String identifier, @QueryParam("depth") final int limit)
```

Comment:
might have some default value the limit?

Reply:
comes with zero by default.